beamer/agent/chain.py

```diff
@@ -779,5 +781,5 @@ def _invalidate(request: Request, claim: Claim, context: Context) -> None:
         request=request.id,
         fill_id=claim.fill_id,
         claim=claim.id,
-        txn_hash=receipt.transactionHash.hex(),
+        txn_hash=receipt.transactionHash.hex(),  # type: ignore
```

Comment:
Eh, I would prefer a way to avoid sprinkling these `# type: ignore` all over.
In general, I like the change to move to `TxReceipt` for `transact`, but I think we should search for a better solution, possibly after merging this.